trace/spanstore.go

```diff
@@ -29,47 +29,8 @@ var (
 	spanStores = make(map[string]*spanStore)
 )
 
-// LatencySampledSpans returns a sample of successful spans.
-//
-// minLatency is the minimum latency of spans to be returned.
-// maxLatency, if nonzero, is the maximum latency of spans to be returned.
-func LatencySampledSpans(name string, minLatency, maxLatency time.Duration) []*SpanData {
-	s := spanStoreForName(name)
-	if s == nil {
-		return nil
-	}
-	var out []*SpanData
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	for i, b := range s.latency {
-		min, max := latencyBucketBounds(i)
-		if i+1 != len(s.latency) && max <= minLatency {
-			continue
-		}
-		if maxLatency != 0 && maxLatency < min {
-			continue
-		}
-		for _, sd := range b.buffer {
-			if sd == nil {
-				break
-			}
-			if minLatency != 0 || maxLatency != 0 {
-				d := sd.EndTime.Sub(sd.StartTime)
-				if d < minLatency {
-					continue
-				}
-				if maxLatency != 0 && d > maxLatency {
-					continue
-				}
-			}
-			out = append(out, sd)
-		}
-	}
-	return out
-}
-
-// ActiveSpans returns the active spans for the given name.
-func ActiveSpans(name string) []*SpanData {
+// ReportActiveSpans returns the active spans for the given name.
+func ReportActiveSpans(name string) []*SpanData {
```

Comment:
Could the Report functions go in another package? I see they are useful for a server status page, but that is really a different use case from adding traces, even though many servers will do both. For example, a client may want traces but will not have a status page.

Reply:
There is a natural cyclic dependency between reporting functions and the trace.SpanData type. If we are not introducing a new type to return from these functions, it is not possible to reorg them to a different package. I really wished that it was possible to split them.

Reply:
I think in a subsequent PR we should explore moving this functionality to an in-memory Exporter.

Reply:
+1.